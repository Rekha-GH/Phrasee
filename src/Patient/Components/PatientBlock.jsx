import React from 'react';
import { useSelector } from 'react-redux';
import { getPatientsWithType } from '../Store/PatientSelector';
import Patient from './Patient';

const PatientBlock = React.memo(({ title, type }) => {
    const list = useSelector(getPatientsWithType(type));

    return (
        <>
            {list.length > 0 && <h4 className="patient-title">{title}</h4>}
            {list.map(patient => <Patient
                name={patient.name}
                joined={patient.joined}
                last_visit_date={patient.last_visit_date}
                is_completed={patient.is_completed}
            />)}
        </>
    );
});

PatientBlock.displayName = 'PatientBlock';
export default PatientBlock;