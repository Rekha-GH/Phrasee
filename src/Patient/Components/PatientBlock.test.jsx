import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import PatientBlock from './PatientBlock';
import { shallow } from '../../setupTests';

describe('PatientBlock Tests', () => {
  const useSelectorMock = jest.spyOn(ReactRedux, 'useSelector');

  test('PatientBlock renders correctly when list is not available', () => {
    useSelectorMock.mockImplementationOnce(() => []); 
    const wrapper = shallow(<PatientBlock title={'A Level'} type='a_level' />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('PatientBlock renders correctly when list is available', () => {
    useSelectorMock.mockImplementationOnce(() => [{
      hospital_id: "56f165b7f21cbefc43efee7bi8yui8uyui8uy",
      is_completed: false,
      joined: "2020-08-06T07:54:00.232Z",
      last_visit_date: "2020-08-06T00:00:00.000Z",
      name: "Adam S",
      type: "a_level", 
      _id: "5f2bb7185756010021aecf228uyhgyuy7ujhy7u"
    }]);

    const wrapper = shallow(<PatientBlock title={'A Level'} type='a_level' />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
