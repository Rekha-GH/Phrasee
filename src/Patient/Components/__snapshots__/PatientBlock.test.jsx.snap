// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientBlock Tests PatientBlock renders correctly when list is available 1`] = `
<Fragment>
  <h4
    className="patient-title"
  >
    A Level
  </h4>
  <Patient
    is_completed={false}
    joined="2020-08-06T07:54:00.232Z"
    last_visit_date="2020-08-06T00:00:00.000Z"
    name="Adam S"
  />
</Fragment>
`;

exports[`PatientBlock Tests PatientBlock renders correctly when list is not available 1`] = `<Fragment />`;
